crates/astria-sequencer/src/accounts/query.rs

```diff
@@ -100,6 +100,11 @@ pub(crate) async fn balance_request(
             };
         }
     };
+
+    // Unstable sort does not allocate auxillary memory and is typically faster. Custom sorting
+    // function is deterministic and only allocates in the case that two values are equal.
```

Comment:
Commented elsewhere: I would very much like to implement `PartialOrd` and `Ord` on `Denom` and its variants instead of doing the allocation here.

Reply:
Done in fb7110f. Also continued down then chain, implementing `PartialOrd` and `Ord` for `TraceSegments` and `PortAndChannel`